src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Octokit;
+
+namespace Microsoft.Docs.Build
+{
+    internal class GitHubAccessor
+    {
+        private readonly GitHubClient _client = new GitHubClient(new ProductHeaderValue("DocFXv3"));
+        private readonly object _syncRoot = new object();
+        private bool _isRateLimitExceeded = false;
+
+        public async Task<(List<Error> errors, UserProfile profile)> GetUserProfileByName(string name)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+
+            var errors = new List<Error>();
+            if (_isRateLimitExceeded)
+                return (errors, null);
+
+            User user;
+            try
+            {
+                user = await _client.User.Get(name);
+            }
+            catch (RateLimitExceededException)
+            {
+                return (ExceedRateLimit(), null);
+            }
+            catch (NotFoundException)
+            {
+                // GitHub will return 404 "Not Found" if the user doesn't exist
+                return (errors, null);
```

Comment:
>errors [](start = 24, length = 6)

do we want to report this warning to user?

Reply:
Yes, it's warned in ContributionInfo.cs

---
In reply to: [207150069](https://github.com/dotnet/docfx/pull/3144#discussion_r207150069) [](ancestors = 207150069)